x/curating/spec/README.md

```diff
@@ -0,0 +1,224 @@
+
+## Messages
+
+### MsgPost
+
+A social media post to be curated is registered on-chain with `MsgPost`. Note that the actual data of the post is not stored, just a hash of the data, and a vendor and post identifier.
+
+Post curation begins as soon as the post is registered on-chain.
+
+```go
+type MsgPost struct {
+	VendorID      uint64 // 1 = twitter, 2 = reddit, etc.
+	PostID        uint64
+	Creator       sdk.AccAddress
+	RewardAccount sdk.AccAddress // optional
+	BodyHash      string // hex string, optional
+	Deposit       sdk.Coin
+}
+```
+
+Posts are persisted with the key `0x01|vendorID|postID`.
+
+An optional reward account enables rewards to go to a different account than the creator. 
+
+Validate the post, calculate the curation end time, and insert the vendor/post_id pair into the post curation queue.
+
+### MsgUpvote
+
+Every transaction on Stakebird includes a deposit which could be slashed by governance for bad behavior. This helps keep the network honest.
+
+The amount of vote credits (`VoteAmount`) is quadratically associated to `VoteNum`, and is used for the upvote amount.
+
+| `VoteNum` | `VoteAmount`  |
+|-----------|---------------|
+|   2       | 4             |
+|   3       | 9             |
+|   4       | 16            |
+
+
+```go
+type MsgUpvote struct {
+	VendorID uint64
+	PostID   uint64
+	Curator  sdk.AccAddress
+	VoteNum	 uint8 // 1 = 1, 2 = 4, 3 = 9 (quadratic)
+	Deposit  sdk.Coin
+}
+```
+
+Upvotes are persisted with the key `0x02|vendorID|postID|curator`.
+
+If the corresponding post doesn't exist yet for an upvote, then also create the post with the curator being the `RewardAccount` for the creator. This way, the curator earns creator rewards as well.
+
+### MsgModerate
+
+`MsgModerate` is to signify spam or illegal content. Moderation delegates to the Cosmos governance module, creating a new text proposal for each moderation message.
+
+[TODO] Figure out how to handle the result of moderation (i.e: re-start curation or delete/flag post). See https://github.com/public-awesome/stakebird/issues/51.
+
+```go
+type MsgModerate struct {
+	VendorID  uint64
+	PostID    uint64
+	Moderator sdk.AccAddress
+	Deposit   sdk.Coin
+}
+```
+
+## End-Block
+
+### Ending Post Curations
+
+Iterate `PostCurationQueue` and process all upvotes that have completed their curation window with the following procedure:
+
+- find all `VPPairs` that have ended curation
+- find all associated `Upvote` objects by iterating the `0x02|vendorID|postID|curator` store
+- calculate the `VotingPool` according to quadratic voting
+- calculate the `MatchedPool` with quadratic finance matching
+- distribute rewards to the creator of the post
+- distribute rewards to each curator
+
+## Events
+
+### EndBlocker
+
+| Type                  | Attribute Key         | Attribute Value           |
+| --------------------- | --------------------- | ------------------------- |
+| complete_curation     | vote_count            | {totalVoteCount}          |
```

Comment:
should these events include info about which post ?

Reply:
Ah yes will add vendor and post ids.